xonsh/environ.py

```diff
@@ -176,7 +176,7 @@ def DEFAULT_ENSURERS():
     'XONSH_DEBUG': (always_false, to_debug, bool_or_int_to_str),
     'XONSH_ENCODING': (is_string, ensure_string, ensure_string),
     'XONSH_ENCODING_ERRORS': (is_string, ensure_string, ensure_string),
-    'XONSH_HISTORY_BACKEND': (is_string, ensure_string, ensure_string),
+    'XONSH_HISTORY_BACKEND': (is_string_or_callable, ensure_string, ensure_string),
```

Comment:
That's right, I don't think it is a huge deal.  But we may as well do the "right" thing while we are thinking about it instead of having a hard to debug issue in the future.